### ISSUE TITLE
error after installing engintron on cpanel

### ISSUE THREAD

**Issue by payamprivate** [engintron/engintron#1135]
Hi ,I just installed Cpanel and then engintron
but nginx cant start , I get this errors in var/log/nginx/error.log  :
Can you help me ? 

2019/12/07 16:17:24 [emerg] 1740#1740: bind() to [::]:443 failed (98: Address already in use)
2019/12/07 16:17:24 [emerg] 1740#1740: bind() to [::]:443 failed (98: Address already in use)
2019/12/07 16:17:24 [emerg] 1740#1740: bind() to [::]:443 failed (98: Address already in use)
2019/12/07 16:17:24 [emerg] 1740#1740: bind() to [::]:443 failed (98: Address already in use)
2019/12/07 16:17:24 [emerg] 1740#1740: bind() to [::]:443 failed (98: Address already in use)
2019/12/07 16:17:24 [emerg] 1740#1740: still could not bind()

**Comment by fevangelou** [comment]
Check that Apache has switched to ports 8080 and 8443 under "Tweak Settings". If not, change them and then do a "Force restart" for Apache/Nginx through the Engintron WHM app.

**Comment by gomcodoctor** [comment]
same issue here, port are changed in Tweak Settings but still port error

**Comment by fevangelou** [comment]
Do a "Force restart" for Apache/Nginx through the Engintron WHM app.

**Comment by gomcodoctor** [comment]
i did, still not working, I am using mod_jk, can it cause issue?

**Comment by fevangelou** [comment]
What's mod_jk?

**Comment by imorandinwnp** [comment]
Hi, try killing nginx first, then restating:

`killall -9 nginx; service nginx restart`

**Comment by fevangelou** [comment]
Sometimes it's best to do `fuser -k 80/tcp; fuser -k 8080/tcp;` to kill stuck connections as well. Restart afterwards both Apache and Nginx.

**Comment by Michael-IDA** [comment]
This is happening on a brand new cPanel installation during the initial install of Engintron.  The only ‘fix’ that worked was the` fuser -k 80/tcp; fuser -k 8080/tcp;` .  

Fotis, please reopen this as a bug report to add stuck connections cleanup to the install and restart scripts.

Best,
Michael

**Comment by Michael-IDA** [comment]
Well, that wiped out Apache :(

Jul 24 14:41:00 {FQDN} restartsrv_httpd[4672]: (98)Address already in use: AH00072: make_sock: could not bind to address [::]:443

> Check that Apache has switched to ports 8080 and 8443 under "Tweak Settings".

Both were listed correctly:
Apache non-SSL IP/port 0.0.0.0:8080
Apache SSL port 0.0.0.0:8443

Had to "Save" "Tweak Settings" before Apache would start.

Install script needs some TLC...

**Comment by Michael-IDA** [comment]
Now I'm getting spammed by:

>  FAILED ⛔: httpd 
> Service check failed to complete
> Unable to connect to port 8080 on 127.0.0.1: Connection refused: Died

**Comment by imorandinwnp** [comment]
@Michael-IDA : check out our script we're using to provision engintron: https://github.com/wnpower/cPanel-Nginx/blob/master/install_engintron.sh
Maybe it will help you

**Comment by Michael-IDA** [comment]
These are the instructions I used off the “Official” install page:

> https://github.com/engintron/engintron#ok-im-sold-how-do-i-install-engintron-on-my-cpanel-server
```
cd /
rm -f engintron.sh
wget --no-check-certificate https://raw.githubusercontent.com/engintron/engintron/master/engintron.sh
bash engintron.sh install
```

Are those instructions wrong?  If so where are the “Official” install instructions?

**Comment by fevangelou** [comment]
@Michael-IDA there's only one official way to install Engintron. What @imorandinwnp mentions is his own deployment script for Engintron (don't forget Engintron is open source...).

As for your issues, services CAN get stuck. And your setup may not exactly be "vanilla" CentOS & cPanel, depending on your hosting company - unless of course it's a VM which you setup from scratch.

In any case, there's a force restart option in Engintron that can correct the port issues, UNLESS for some reason Apache's ports did not change by the install script, which simply edits Tweak Settings from the command line (so to speak).

I've seen Apache break from other stuff as well, e.g. bad inclusions to Apache (pre/global).

It would also help if you let us know what CentOS version you're using (I hope it's not 8 as not even cPanel is tested for it)...

**Comment by Michael-IDA** [comment]
Hi Fotis,

Thanks for the reply.  And yes, I should have included spec’s, sorry about that.

Virtuozzo 7
CentOS 7.8.2003 
Linux {fqdn} 3.10.0 #1 SMP Mon Dec 9 19:36:21 MSK 2019 x86_64 x86_64 x86_64 GNU/Linux
Intel(R) Xeon(R) CPU E5-2630 v4 @ 2.20GHz
RAM:  12GB
Disk:  60GB
cPanel: v88.0.13 
EasyApache 4:  Apache 2.4, PHP 7.2, and PHP 7.3 


I was able to install a CMS (Backdrop) through its web interface, so doing several manual kills and forced restarts seemed to have eventually worked.  I did several of each and didn’t start the CMS install until `systemctl status` indicated both httpd.service and nginx.service were successfully running. [1]

Systemd is notorious for slow process starts/kills.  Basically what I was requesting by ‘TLC’ [2] above is that the install, force restart, and (anywhere else it makes sense) scripts have added in appropriate spots something like:

- ‘sleep’ waits after both kills and starts, and/or
- ‘Did the kill finish yet?’ logic, and/or
- `fuser -k 80/tcp; fuser -k 8080/tcp;` before starts (add 443 as well?).

Basically something to catch the expected hiccups so new users aren’t immediately turned off.

Best Regards,

Michael
Internet Design Alliance, owner
http://inet-design.com/
Fully Managed Drupal and Backdrop CMS Services

[1] cPanel’s email bombing quit after awhile.
[2] Americanism: Tender Loving Care

**Comment by fevangelou** [comment]
Normally you wouldn't need any pauses between these restart actions. So it's weird this occurs on your side. It *may* relate to Apache getting stuck though, so try regular restarts from the terminal (execute /engintron.sh to see all available actions) and see if Apache is taking long to restart (the Engintron WHM app will wait for the entire process and then output everything in an instant).

As for Backdrop (which is a Drupal fork), why does not it need to be "started" as you mention? Does it deploy differently to Drupal?
